### ISSUE TITLE
unit-test-quality: 'Awaiting non-promise values' regex flags three-level calls and line-wrapped chains

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#84]
**Version:** 7.19.0
**Law:** `unit-test-quality-standards` (async-test-analyzer)

## What happens

`Awaiting non-promise values detected` fires on code that awaits a promise.
Every instance in our repo is a false positive, from two distinct gaps in the
same regex:

```js
// async-test-analyzer-configuration.js
/await\s+(?!\w+\(|\w+\.\w+\(|new\s+Promise)[\w.]+(?!\()/g
```

### 1. The lookahead stops at two levels

It excludes `await fn(` and `await obj.fn(`, but not `await a.b.c(`:

```ts
await page.keyboard.press('Escape');   // flagged
await page.goto('/laws');              // not flagged
```

Any three-level call is reported. That is common in Playwright, and in any
fluent API.

### 2. A chain broken across lines is read as a bare identifier

Prettier wraps long chains. The regex sees `await expect` with nothing after
it on that line:

```ts
await expect              // flagged
  .poll(() => page.evaluate(() => document.documentElement.scrollWidth))
  .toBeLessThanOrEqual(390);
```

The same expression on one line is not flagged, so the finding depends on line
width rather than on what the code does.

## Impact

11 matches in our e2e suite, 0 of them real. They land in
`unit-test-quality-standards`, which is one of the larger contributors to our
visible debt, so the number reads as test-quality debt that cannot be paid —
the only way to satisfy it is to stop using a fluent API or to fight the
formatter.

## Suggested direction

Match the whole expression rather than the token after `await`: allow any
member chain ending in a call, and let the match span newlines (`[\s\S]`) so a
wrapped chain is read as one expression. A cheaper first cut is to allow N
levels — `(?!(?:\w+\.)*\w+\()` — which closes case 1 on its own.

## Also, a message that names something it never checked

`HTTP calls should be mocked in unit tests` fires from
`test-coverage-analyzer` when a file matches `DEPENDENCY_PATTERNS`
(`import ... from`, `require(`, `inject(`, `TestBed`) and no
`MOCK_PATTERNS`. There is no HTTP check anywhere in that path — a spec with a
single `import` and no mock is reported for HTTP calls it does not make.

The mock requirement itself is #29; this is only about the wording, which sends
the reader looking for network code that is not there.

**Comment by ruleofcodedev** [comment]
Fixed in #87, on `develop`. Both halves.

## The await pattern

Narrowed to **literals** — a number, a string, a boolean, `null`/`undefined`, or an array where `Promise.all` was meant. That is the whole set this analyzer can be right about.

Your two cases and a third you did not list:

| code | before | after |
| --- | --- | --- |
| `await page.keyboard.press('Escape')` | reported | silent |
| `await expect` + wrapped `.poll(…)` | reported | silent |
| `await promiseVariable` | reported | silent |
| `await 5` / `await [p1, p2]` | reported | still reported |

We did **not** take the cheaper first cut you suggested (`(?!(?:\w+\.)*\w+\()`). It closes case 1, but `await promiseVariable` — correct code, and indistinguishable from the mistake without type information — would have stayed. Object literals are deliberately excluded too: `await { then() {} }` awaits a thenable and is legitimate.

Your framing of the cost is the reason this went in this batch rather than later: satisfiable only by abandoning a fluent API or fighting the formatter is not debt, and it sat inside a law that carries real debt, which made the real number unreadable.

## The message that named something it never checked

You are right about the effect and we owe you a correction on the mechanism.

It does not come from `DEPENDENCY_PATTERNS` — that branch prints `Dependencies found but no mocking detected in {file}`, a different message. The HTTP finding has its own checker, and it was `content.includes('http')` over the whole file text. Verified by fixture rather than by reading:

```
// see https://example.com/docs   →  "HTTP calls should be mocked in unit tests"
```

So a URL in a comment, in a string, or in an import path containing `http` all triggered it. Same conclusion you reached, reached a different way. It requires an actual call now — `fetch(`, `axios.post(`, `http.get<T>(`, `XMLHttpRequest` — and stays silent once the request is mocked.

The mock requirement itself is still #29 and still open.
